config/zfs-build.m4

```diff
@@ -164,6 +164,7 @@ AC_DEFUN([ZFS_AC_CONFIG_ALWAYS], [
 	ZFS_AC_CONFIG_ALWAYS_TOOLCHAIN_SIMD
 	ZFS_AC_CONFIG_ALWAYS_SYSTEM
 	ZFS_AC_CONFIG_ALWAYS_ARCH
+	ZFS_AC_CONFIG_ALWAYS_PAM
```

Comment:
I think instead of ALWAYS we probably only would need this for USER (ZFS_AC_CONFIG_USER in config/user.m4), and probably only for Linux unless you intend to have it working on FreeBSD too.

Reply:
I did a bit of basic testing and it seems with a bit of import fixing the pam module seems to work fine on freebsd, too. I just imported a different header and supplied a dummy log function and it worked fine. Regarding user vs. kernel vs. "always". I remember I had it in user before, but there were problems, but maybe that was just me not getting along with autoconf.